providers/implementations/digests/build.info

```diff
@@ -15,7 +15,7 @@ $MD2_GOAL=../../liblegacy.a
 $MD4_GOAL=../../liblegacy.a
 $MDC2_GOAL=../../liblegacy.a
 $WHIRLPOOL_GOAL=../../liblegacy.a
-$RIPEMD_GOAL=../../liblegacy.a
+$RIPEMD_GOAL=$COMMON_GOAL
```

Comment:
You might want to add a comment here, saying something along the lines of:

```
# An implementation is usually added in the library of the provider where it's used.
# RIPEMD is unusual in that it's provided by both the legacy provider and the default
# provider, and to avoid all sorts of clashes, we place the implementation in a library
# that's common to both.
```

Reply:
It also of course means the ripemd source files will be counted as sources for the fips provider and included in the checksums.

Reply:
See the diff checksums output here https://github.com/openssl/openssl/actions/runs/3225036540/jobs/5276852073